svm/src/transaction_processor.rs

```diff
@@ -498,7 +480,22 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             fee_details.total_fee(),
         )?;
 
-        Ok((fee_details, fee_payer_account, fee_payer_rent_debit))
+        // Capture fee-subtracted fee payer account and original nonce account state
+        // to rollback to if transaction execution fails.
+        let rollback_accounts = RollbackAccounts::new(
+            nonce,
+            *message.fee_payer(),
```

Comment:
nit: seems we can use a local variable called `fee_payer_address`, instead of another method call of `.fee_payer()`.